ibtk/include/ibtk/FEDataManager.h

```diff
@@ -160,6 +160,26 @@ class FEDataManager : public SAMRAI::tbox::Serializable, public SAMRAI::mesh::St
             return;
         }
 
+        /*!
+         * Alternative indexing operation: retrieve all dof indices of all
+         * variables in the given system at once by reference.
+         */
+        inline
+        const std::vector<std::vector<libMesh::dof_id_type>>&
+        dof_indices(const libMesh::Elem* const elem)
+        {
+            std::vector<std::vector<libMesh::dof_id_type>>& elem_dof_indices = d_dof_cache[elem->id()];
+            if (elem_dof_indices.size() == 0)
```

Comment:
Yes, definitely cleaner than my approach.

However, I think that the size can be between 0 and the number of variables thanks to the original interface. I think that case also needs to change to look up all at once for this to work robustly.

Reply:
Good point: that needs to be fixed for both methods to coexist.